src/Http/Http.Features/src/IHttpRequestFeature.cs

```diff
@@ -14,6 +14,16 @@ public interface IHttpRequestFeature
     /// </summary>
     string Protocol { get; set; }
 
+    /// <summary>
+    /// Gets or sets the :protocol field used with the CONNECT method.
+    /// See https://datatracker.ietf.org/doc/html/rfc8441.
+    /// </summary>
+    string? ConnectProtocol
```

Comment:
I want to go back in time and challenge why we made these all settable on the interface 😄